contrib/dhcp_agent/dhcp_agent.py

```diff
@@ -0,0 +1,350 @@
+
+    def fetch_and_reload(self, mode, request_params):
+        dhcp_config = self._get_configuration(mode, request_params)
+        if dhcp_config:
+            self._set_new_configuration(dhcp_config, mode)
+            self._restart_dhcp_server()
+
+    def get_request_params(self, dcs, envs):
+        request_params = []
+        if envs:
+            request_params.extend([('env', env) for env in envs])
+        if dcs:
+            request_params.extend([('dc', dc) for dc in dcs])
+        assert request_params != []
+        return request_params
+
+    def _get_configuration(self, mode, params):
+        url = '{}://{}/dhcp/{}/?{}'.format(
+            self.proto,
+            self.host,
+            mode,
+            urlencode(params)
+        )
+        data = None
+
+        self.logger.info('Sending request to {}'.format(url))
+        try:
+            response = self.make_request(url)
+        except HTTPError as e:
+            if e.code != 304:
+                self.logger.error(
+                    'Server returned {} status code with message "{}"'.format(
+                        e.code, e.fp.read().decode()
+                    )
+                )
+                return False
+            else:
+                self.logger.info(
+                    'Server return status 304 NOT MODIFIED. Nothing to do.'
+                )
+                return
+        else:
+            data = response.read()
+            self.set_last_modified(url, response.headers.get('Last-Modified'))
+        return data
+
+    def _send_confirm(self):
+        url = '{}://{}/dhcp/synch/'.format(
+            self.proto,
+            self.host,
+        )
+        try:
+            self.make_request(url, data={})
+        except HTTPError as e:
+            self.logger.error(
+                'Could not send confirmation to Ralph. '
+                'Server returned {} status code with message: {}'.format(
+                    e.code, e.fp.read().decode()
+                ),
+            )
+            return False
+        self.logger.info('Confirmation sent to {}.'.format(self.host))
+        return True
+
+    def _set_new_configuration(self, config, mode):
+        config_path = getattr(self, 'dhcp_{}_config_path'.format(mode))
+        try:
+            if config_path:
+                with open(config_path, 'w') as f:
+                    f.write(config)
+                self.logger.info(
+                    'Configuration written to {}'.format(config_path)
+                )
+            else:
+                sys.stdout.write(config.decode())
+                self.logger.info('Configuration written to stdout.')
+            return True
+        except IOError as e:
+            self.logger.error(
+                'Could not write new DHCP configuration. Error '
+                'message: {}'.format(e),
+            )
+            return False
+
+    def _restart_dhcp_server(self):
```

Comment:
This function returns either None (line 319) or True/False (line 337). What about returning only True or False..?